neutrino.go

```diff
@@ -1078,6 +1078,32 @@ func (s *ChainService) BanPeer(addr string, reason banman.Reason) error {
 	return s.banStore.BanIPNet(ipNet, reason, BanDuration)
 }
 
+// UnBanPeer connects and unban's a previously banned peer.
+func (s *ChainService) UnBanPeer(addr string) error {
+	log.Warnf("UnBanning peer %v", addr)
+
+	defer func() {
+		if sp := s.PeerByAddr(addr); sp != nil {
+			tcpAddr, err := s.addrStringToNetAddr(addr)
+			if err != nil {
+				log.Warnf("unable to lookup IP for "+
+					"%v: %v", addr, err)
+			}
```

Comment:
I think we should return when we get an error so we do not add it to the `connManager`

Reply:
Thank you so much for this. Fixed it.